### ISSUE TITLE
``torch.split`` followed by ``torch.cat`` fails to restore the tensor (v2.1-2.5)

### ISSUE THREAD

**Issue by jeffhataws** [pytorch/xla#8410]
## 🐛 Bug

``torch.split`` is used to split a multi-dim tensor into chunks. Then ``torch.cat`` is used to put the shards back. However, the restored tensor content is different from the original tensor.

Fails for v2.1 to v2.5 and passes for v2.6 (top-of-tree). Documenting this issue here as a known issue for v2.1-v2.5.

## To Reproduce

Save the following code as test_split.py:

```
import torch
import torch_xla
import torch_xla.core.xla_model as xm

def test_split():
    x = torch.rand([1, 2048, (1536 + 128 + 128)])
    a,b,c = torch.split(x, [1536, 128, 128], dim=-1)
    xx = torch.cat([a,b,c], dim=-1)
    assert (x == xx).all()

    x_xla = x.to('xla')
    aa,bb,cc = torch.split(x_xla, [1536, 128, 128], dim=-1)
    xx_xla = torch.cat([aa,bb,cc], dim=-1)

    print(x)
    print(xx_xla.cpu())
    assert (x_xla.cpu() == xx_xla.cpu()).all()

    print("All checks passed!")

```

PJRT_DEVICE=CPU pytest test_split.py

## Expected behavior

Expect the ``torch.split`` followed by ``torch.cat`` to restore the tensor contents to same locations.

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]:
 - torch_xla version: 2.1-2.5

**Comment by jeffhataws** [comment]
Let's add this test to the testsuite.

**Comment by ManfeiBai** [comment]
Thanks, and let's assign to @tengyifei who will help 2.6 release

**Comment by ysiraichi** [comment]
I can't seem to reproduce this issue.
I have tried on versions: 2.7.0 (nightly) and on 2.5.0.
@jeffhataws Could you try the nightly builds?

**Comment by jeffhataws** [comment]
> I can't seem to reproduce this issue. I have tried on versions: 2.7.0 (nightly) and on 2.5.0. [@jeffhataws](https://github.com/jeffhataws) Could you try the nightly builds?

It is fixed in 2.6. Just need to add a testcase @ysiraichi .

**Comment by ysiraichi** [comment]
Oh, I see. Sorry, I missed that info in the issue description.
